pkgs/servers/misc/lms/default.nix

```diff
@@ -0,0 +1,96 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, writeText
+, cmake
+, pkg-config
+, gtest
+, boost
+, zlib
+, ffmpeg_4
+, graphicsmagick
+, libconfig
+, taglib
+, wt, useMinimalWt ? false
+}:
+
+let
+  ffmpeg = ffmpeg_4;
+
+  # Wt with the minimum set of optional dependencies required for LMS
+  wt-minimal = wt.override {
+    # Docs not needed
+    doxygen = null;
+    # Graphics libraries not needed
+    glew = null; libharu = null; pango = null; graphicsmagick = null;
+    # LMS uses only sqlite
+    firebird = null; libmysqlclient = null; postgresql = null;
+    # Qt not used
+    qt48Full = null;
```

Comment:
I'm not sure what better way there is. 1) The wt derivation (in `development/libraries/wt/default.nix`) does not provide a nicer mechanism to remove components and 2) I'm don't think that the wt derivation should be expanded, given that wt simply does not build the [optional components](https://www.webtoolkit.eu/wt/doc/reference/html/InstallationUnix.html) if it doesn't find the required dependencies, and that cutting out optional dependencies is likely only useful on a system that isn't served by a binary cache (e.g. armv7l, on which those dependencies are broken half the time). Having the caller pass in null seems like a robust and minimal way to include or drop those optional portions.

Or, are you in against providing the "useMinimalWt" option at all? I'm happy to remove that and keep that in my overlays; I put that in because I thought others might find it useful.